include/drivers/hwinfo.h

```diff
@@ -39,7 +39,8 @@ extern "C" {
  * @param buffer  Buffer to write the ID to.
  * @param length  Max length of the buffer.
  *
- * @retval size of the device ID copied or negative on error.
+ * @retval size of the device ID copied
+ * @retval -ENOTSUP if there is no implementation for the particular device.
```

Comment:
`size of the device ID copied or negative on error` is clear defined behavior, but not `-ENOTSUP if there is no implementation for the particular device`, what about other return values?

Reply:
There are no other errors.

Reply:
> There are no other errors.

What is the reason to change it? Is it sure that future implementations would not need to return anything else as -ENOTSUP? What about an internal error like -EIO or -EBUSY?

Reply:
I'd like to declare the API as stable, and the requirement from the API meeting was that I have to specify which errors could be returned. If we have another driver that could return another error, we have to announce this change, I guess.

Reply:
This PR started as doc-change and escalated quickly to 8 commits and +128 lines.

Reply:
I just do not see the reason to change clear defined `retval`.

Reply:
I was not in the meeting. I don‘t know what was discussed there. @cales might know.

Reply:
From dev-review meeting: Document the specific error for -ENOTSUP and than document that any other negative value would be a driver specific error.